common/djangoapps/student/views.py

```diff
@@ -2717,3 +2717,23 @@ def get_context_data(self, **kwargs):
         })
 
         return context
+
+
+class LogoutRedirectView(TemplateView):
+    """
+    Logs out user, performing the same logic as LogoutView, and redirects the user to a specified target.
+    """
+
+    template_name = 'logout.html'
+
+    def get_context_data(self, **kwargs):
+        context = super(LogoutRedirectView, self).get_context_data(**kwargs)
+        target = kwargs.get('redirect_to', '/')
+        logout_uri = "{}?{}".format(reverse('logout'), 'no_redirect=1')
+
+        context.update({
+            'target': target,
+            'logout_uri': logout_uri,
```

Comment:
Haha yea, it was working yesterday when I left and then when I took a look again today, I saw the same thing you did and was confused... so I pushed some updates but have not tested them because my devstack/vagrant decided to poop out on me.